src/test/java/it/eng/idsa/businesslogic/service/impl/ClearingHouseServiceImplTest.java

```diff
@@ -1,45 +1,69 @@
 package it.eng.idsa.businesslogic.service.impl;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
 
-import org.junit.Before;
+import javax.xml.datatype.DatatypeConfigurationException;
 
-import de.fraunhofer.iais.eis.Message;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.web.client.RestTemplate;
+
+import de.fraunhofer.iais.eis.util.ConstraintViolationException;
+import it.eng.idsa.businesslogic.configuration.ApplicationConfiguration;
+import it.eng.idsa.businesslogic.processor.exception.ExceptionForProcessor;
+import it.eng.idsa.businesslogic.service.HashFileService;
+import it.eng.idsa.businesslogic.util.TestUtilMessageService;
 
 public class ClearingHouseServiceImplTest {
 	
-	ClearingHouseServiceImpl clearingHouseServiceImpl = new ClearingHouseServiceImpl();
+	@InjectMocks
+	private ClearingHouseServiceImpl clearingHouseServiceImpl;
+	
+	private String payload;
 	
-	MultipartMessageServiceImpl multipartMessageServiceImpl = new MultipartMessageServiceImpl();
+	@Mock
+	private HashFileService hashService;
 	
-	Message message;
+	@Mock
+	private RestTemplate restTemplate;
 	
-	String payload;
+	@Mock
+	private ApplicationConfiguration configuration;
 	
-	String directory = "./src/test/resources/ClearingHouseServiceImplTest/";
+	String mockEndpoint;
 	
 	
-	@Before
+	@BeforeEach
 	public void init() {
-		String clearingHouseHeader = null;
-		String clearingHousePayload = null;
-		try {
-			clearingHouseHeader = new String(Files.readAllBytes(Paths.get(directory+"clearingHouseHeader.txt")));
-			clearingHousePayload = new String(Files.readAllBytes(Paths.get(directory+"clearingHousePayload.txt")));
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
-		message = multipartMessageServiceImpl.getMessage(clearingHouseHeader);
-		payload = clearingHousePayload;
+		MockitoAnnotations.initMocks(this);
+		payload = "{\"foo\":\"bar\"}";
+		mockEndpoint = "https://clearinghouse.com";
+		when(hashService.hash(payload)).thenReturn("ABC");
+		when(configuration.getClearingHouseUrl()).thenReturn(mockEndpoint);
+		when(restTemplate.postForObject(any(), any(), any())).thenReturn(null);
 	}
 	
-	/*
-	 * @Test public void testRegisterTransaction () {
-	 * assertFalse(clearingHouseServiceImpl.registerTransaction(message, payload));
-	 * }
-	 */
+	
+	  @Test
+	  @Disabled
+	  //TODO enable test when CH is on the correct infomodel
+	  public void testRegisterTransactionFail () {
```

Comment:
@davidjovanovic 
Why is this service disabled, can we enable it, since we now have enabled CH.
What do we verify in this test?

Reply:
Enabled the test. It now verifies if the method returns false and if a rejection is sent.